Sources/Services/AWSSTS/STSClient.swift

```diff
@@ -462,3 +462,12 @@ extension STSClient: STSClientProtocol {
     }
 
 }
+
+extension STSClient {
+    public func presignRequestForGetCallerIdentity(input: GetCallerIdentityInput, expiration: Foundation.TimeInterval) async throws -> URLRequest {
+        guard let presignedRequest = try await input.presign(config: config, expiration: expiration) else {
+            throw ClientError.unknownError("Returned request from input.presign() was nil for the operation GetCallerIdentity.")
+        }
+        return try await presignedRequest.toURLRequest()
```

Comment:
`toURLRequest()` is not a Swift-convention way to make such a conversion. `to...` is more of a Java-land convention.

I would prefer extending `URLRequest` with a custom initializer, i.e.
```
return try await URLRequest(sdkHttpRequest: presignedRequest)
```

Reply:
I see, but I think it makes sense to keep it this way given this is the convention used in `SdkHttpRequest`, as shown here: https://github.com/smithy-lang/smithy-swift/blob/6037a8157082d00da6ff1be1462022dcaccb3662/Sources/ClientRuntime/Networking/Http/SdkHttpRequest.swift#L60-L86

Reply:
We should reduce the amount of code that doesn't follow Swift convention rather than increase it.  The code you cite should not have been named that way either.

Reply:
Fair, I'll create extension and put the changes up.

Reply:
I agree with Josh let's be as Swifty as possible rather than rely on previous code examples :) Especially true as we deprecate older swift versions and gain access to newer Swift syntax features!

Reply:
I don't know if it was written by developers not familiar with Swift convention or just borrowed from elsewhere without adapting it to Swift, but you will see some code (like this) in our projects that doesn't follow Swift community convention.  We don't need to (can't!) change it all overnight, but we do need to chip away at it when the opportunity presents itself.